extensions/resteasy-reactive/rest-client-reactive/deployment/src/main/java/io/quarkus/rest/client/reactive/deployment/RestClientReactiveProcessor.java

```diff
@@ -580,6 +573,28 @@ && isImplementorOf(index, target.asClass(), RESPONSE_EXCEPTION_MAPPER, Set.of(AP
         }
     }
 
+    /**
+     * Based on a list of interfaces implemented by @Provider class, determine if registration
+     * should be skipped or not. Server-specific types should be omitted unless implementation
+     * of a <code>ClientRequestFilter</code> exists on the same class explicitly. 
+     * Features should always be omitted.
+     */
+    private boolean skipAutoDiscovery(List<DotName> providerInterfaceNames) {
+        if (providerInterfaceNames.contains(ResteasyReactiveDotNames.FEATURE)) {
+            return true; 
+        }
+        if (providerInterfaceNames.contains(ResteasyReactiveDotNames.CONTAINER_REQUEST_FILTER)
+                || providerInterfaceNames.contains(ResteasyReactiveDotNames.CONTAINER_RESPONSE_FILTER)
+                || providerInterfaceNames.contains(ResteasyReactiveDotNames.EXCEPTION_MAPPER)) {
+            if (providerInterfaceNames.contains(DotName.createSimple(ClientRequestFilter.class.getName()))) {
```

Comment:
This is missing `ClientResponseFilter`.
Also, I think both the DotNames should be constants of this class (if not defined elsewhere) - see existing exampes.

Btw, I was thinking about checking only for the package prefix `jakarta.ws.rs.client.`. This way we'd auto-catch future jakarta interfaces (if any). But I'm a bit on the fence, checking the two interfaces explicitly is probably fine.

Reply:
Now using DotName constants and also checking for response filter.